### DISCUSSION TITLE
T-SQL и DML, управляем данными

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/LessonMonitor#79] [MSSQL]
Для выполнения домашней работы, вам необходимо открыть и сделать форк основного репозитория по ссылке:
https://github.com/cleannetcode/LessonMonitor/tree/homework/sql%2F4

По выполнению работы, вам необходимо создать пулл реквест из вашей ветки с выполненной домашкой в основной репозиторий.

Почитать:
https://www.sqlservertutorial.net/sql-server-basics - Section 1-6, 11 (чтение, группировка и модификация данных)

Сделать:
- Создать базу с таблицами (Таймкода, Вопросы, Уроки, Участники)
- Настроить связи между таблицами
- Заполнить таблица данными
- Попробовать удалить и отредактировать данные
- Подготовить селект запросы, которые будут выдавать следующее:
1) Имя участника, Название урока, кол-во вопросов, кол-во таймкодов, кол-во посещенных занятий
1.1) - сортировка по дате урока
2*) Собрать таймкода для конкретного занятия (фильтруем по LessonId): Время таймкода - название 
- попробовать использовать для получившихся SELECT запросов, команды (IN, DISTINCT, GROUP BY, ...)

Проверить:
провести ревью кода любого чужого пулл реквеста относящегося к данной домашке.

**Comment by pingvin1308** [comment]
Попробуйте сделать вывод Имя участника, Название урока, кол-во вопросов, кол-во таймкодов, кол-во посещенных занятий
Одним запросом, чтобы получилась одна таблица

**Reply by AshtonBro** [reply]
### Выполнил вывод **статистики** в одну таблиц. <h3>

**Переджойнивание таблиц не дал мне тот результат который я хотел увидить, в итоге, пришел к более менее элегантному** **решению и собрал, так сказать __"View-шку"__ с читабельными данными копируя блоки таблиц во временные таблицы и далее обьединил в одну.**

* Временная таблица №1
![image](https://user-images.githubusercontent.com/54149866/120380144-39ca2280-c33a-11eb-89e9-28852418add0.png)

* Временная таблица №2
![image](https://user-images.githubusercontent.com/54149866/120380217-55cdc400-c33a-11eb-864a-48ae50cde34c.png)

* Временная таблица №3
![image](https://user-images.githubusercontent.com/54149866/120380271-6716d080-c33a-11eb-8013-db7eceaa16e2.png)

* Основной запрос, где мы через **JOIN** соединяем всё в едино и выводим запрос.
![image](https://user-images.githubusercontent.com/54149866/120380468-a513f480-c33a-11eb-8e98-a1299ad3ee24.png)

* Получаем результат
     * Имя участника,
     *  Почта, 
     * Кол-во посещенных занятий учеником,
     * Кол-во вопросов заданные учеником (без привязки к уроку),
     * Название урока, 
     * Кол-во таймкодов
     

[СЫЛКА НА СКРИПТ, прокрутить вниз](https://github.com/AshtonBro/LessonMonitor/blob/homework/sql/4/LessonMonitor/LessonMonitor.SqlScripts/LessonDb.sql)
Данные некоторые дублируются, но у нас есть в целом продовая информацию которую не стыдно отобразить на front'е.
![image](https://user-images.githubusercontent.com/54149866/120380660-e5737280-c33a-11eb-971f-7c2d201c12f2.png)
